eth_account/local.py

```diff
@@ -59,3 +59,9 @@ def signTransaction(self, transaction_dict):
 
     def __bytes__(self):
         return self.privateKey
+
+    def __hash__(self):
+        return hash(self.privateKey)
+
+    def __eq__(self, other):
+        return (isinstance(other, self.__class__)) and (self.__hash__ == other.__hash__)
```

Comment:
Instead of `self.__hash__ == other.__hash__`, try something like `self.privateKey == other.privateKey`.

Reply:
Also, I think `type(self)` is preferred over `self.__class__`.  Same with `hash(self)` over `self.__hash__`, largely to avoid private attribute access.

Reply:
@carver  @pipermerriam  Thanks for the feedback! I'll look into getting this switched.